css/css-transitions-1/events-007.html

```diff
@@ -1,64 +1,50 @@
-    <meta name="flags" content="dom">
-    <meta name="assert" content="Making an element display:none; while it has a transition in progress should prevent a transitionend event from getting fired.">
-    <script src="/resources/testharness.js"></script>
-    <script src="/resources/testharnessreport.js"></script>
-    <style>
-#a {
-    height: 100px;
-    width: 100px;
-    transition: background-color 2s;
-}
-.red {
-    background-color: red;
-}
-.blue {
-    background-color: blue;
-}
-.hidden {
-    display: none !important;
-}
-    </style>
-    <script>
-async_test(function (t) {
-    window.addEventListener('load', function () {
-        var div = document.getElementById('a');
-        var ended = false;
-        var after = false;
-        div.addEventListener('transitionend', function () {
-            ended = true;
-            t.step(function () {
-                assert_false(true, "transitionend event didn't fire");
-            });
-        }, false);
+<meta charset="utf-8">
+<title>CSS Transitions Test: no transitionend event after display:none</title>
+<link rel="author" title="Chris Rebert" href="http://chrisrebert.com">
+<link rel="help" href="http://www.w3.org/TR/css3-transitions/#transition-events">
+<link rel="help" href="https://lists.w3.org/Archives/Public/www-style/2015Apr/0405.html" title="[CSSWG] Minutes Telecon 2015-04-29" data-section-title="AnimationEnd events and display: none">
+<meta name="assert" content="Making an element display:none; while it has a transition in progress should prevent a transitionend event from getting fired.">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="./support/helper.js" type="text/javascript"></script>
 
-        div.className = 'blue';// initiate transition
-        window.setTimeout(function () {
-            t.step(function () {
-                assert_false(ended, "transitionend did not fire before hiding callback ran");
-                assert_false(after, "hiding callback ran before the end of the test");
-            });
-            div.className += ' hidden';// force display:none during the transition
-        }, 1000);// halfway into the transition
-        window.setTimeout(function () {
-            after = true;
-            t.step(function () {
-                assert_false(ended, "transitionend event didn't fire");
-                t.done();
-            });
-        }, 2100);// after the transition would have ended
-    }, false);
-}, "transitionend should not be fired if the element is made display:none during the transition");
-    </script>
 </head>
 <body>
-    <div id="a" class="red"></div>
+
+<div id="log"></div>
+
+<script>
+async_test(t => {
+  window.addEventListener('load', function () {
+    const div = addDiv(t, { 'style': 'transition: background-color 0.5s;' +
```

Comment:
I suspect this should be less than 0.5s, e.g. 0.2s, so the test would usually fail if "div.style.display = 'none';" was not present.

Reply:
Hmm, I did check that the test fails without `div.style.display = 'none';` at least in Gecko (I ran all the other tests in Chrome too but didn't check this one failed without the `display:none` in Chrome). The reason this should work is we don't start the timeout until the `requestAnimationFrame` callback by which point the transition should be running so we should have <500ms until the transition ends. We can make the transition shorter but then we increase the chance of the `requestAnimationFrame` running after the transition has already completed.

How about I make the transition `0.4s` just in case there's a UA that doesn't start the transition until the `requestAnimationFrame`.

Reply:
Sure